pkg/context/context.go

```diff
@@ -68,15 +71,26 @@ func NewOrDie(ctx cloudprovider.Context) Context {
 		logging.FromContext(ctx).Fatalf("Checking EC2 API connectivity, %s", err)
 	}
 	logging.FromContext(ctx).With("region", *sess.Config.Region).Debugf("discovered region")
+
 	subnetProvider := subnet.NewProvider(ec2api)
 	securityGroupProvider := securitygroup.NewProvider(ec2api)
+	pricingProvider := pricing.NewProvider(
+		ctx,
+		pricing.NewAPI(sess, *sess.Config.Region),
+		ec2api,
+		*sess.Config.Region,
+		settings.FromContext(ctx).IsolatedVPC,
```

Comment:
Can we remove this from the constructor since we can just grab this from the context?